powershell-adapter/psDscAdapter/win_psDscAdapter.psm1

```diff
@@ -584,6 +588,72 @@ function ValidateMethod {
     return $method
 }
 
+function GetClassBasedProperties {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string] $filePath,
+
+        [Parameter(Mandatory = $true)]
+        [string] $className
+    )
+
+    if (".psd1" -notcontains ([System.IO.Path]::GetExtension($filePath))) {
+        return @('get', 'set', 'test')
+    }
+
+    $module = $filePath.Replace('.psd1', '.psm1')
```

Comment:
Although probably true in most cases, it seems we should be using `RootModule` in the psd1 instead of assuming the psm1 has same name